### PR TITLE
Use Handle API 

### PR DESCRIPTION
**Pull request by southeo**
Creates PID records using the Handle API. 

RFC for the PID record is in development. Currently, the mediaObject PID records include the following information: 

| FDO Attribute | Source in media object |
|------------------- | ------------------------------ |
| fdoProfile | https://hdl.handle.net/21.T11148/64396cf36b976ad08267 (placeholder value) |
| digitalObjectType | https://hdl.handle.net/21.T11148/64396cf36b976ad08267 (placeholder value) |
| issuedForAgent | https://ror.org/0566bfb96 (Naturalis ROR) |
| referentName | `mediaObject.type() for mediaObject.digitalSpecimenId()` |
| mediaUrl | `mediaUrl` from `mediaObject.attributes()` |
| subjectId | `mediaObject.digitalSpecimenId()` |

### CODE REVIEW

File: src/test/java/eu/dissco/core/digitalmediaobjectprocessor/web/HandleComponentTest.java [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+  private JsonNode givenHandleRequest() throws Exception{
+    return MAPPER.readTree("""
+        {
+        "data": {
+            "type": "mediaObject",
+            "attributes": {
+              "fdoProfile": "https://hdl.handle.net/21.T11148/64396cf36b976ad08267",
+              "digitalObjectType": "https://hdl.handle.net/21.T11148/64396cf36b976ad08267",
+              "issuedForAgent": "https://ror.org/0566bfb96",
+              "mediaHash":"",
```
Comment (samleeflang) [inline src/test/java/eu/dissco/core/digitalmediaobjectprocessor/web/HandleComponentTest.java:257]: I think these also need to be updated and remove the removed fields

File: src/test/java/eu/dissco/core/digitalmediaobjectprocessor/service/FdoRecordServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    return result;
+  }
+
+  private static JsonNode givenPostAttributes() {
+    var attributes = MAPPER.createObjectNode();
+    attributes.put(FDO_PROFILE.getAttribute(), FDO_PROFILE.getDefaultValue());
+    attributes.put(DIGITAL_OBJECT_TYPE.getAttribute(), DIGITAL_OBJECT_TYPE.getDefaultValue());
+    attributes.put(ISSUED_FOR_AGENT.getAttribute(), ISSUED_FOR_AGENT.getDefaultValue());
+    attributes.put(REFERENT_NAME.getAttribute(), TYPE + " for " + DIGITAL_SPECIMEN_ID);
+    attributes.put("mediaHash", "");
```
Comment (samleeflang) [inline src/test/java/eu/dissco/core/digitalmediaobjectprocessor/service/FdoRecordServiceTest.java:140]: Remove, removed fields?

**Review by samleeflang — COMMENTED** [review]
Small comments, looking great otherwise!

File: src/main/java/eu/dissco/core/digitalmediaobjectprocessor/domain/FdoProfileAttributes.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package eu.dissco.core.digitalmediaobjectprocessor.domain;
+
+public enum FdoProfileAttributes {
+
+  TYPE("type", "mediaObject"),
```
[the remark this answers is not in the document]
Comment (southeo) [reply src/main/java/eu/dissco/core/digitalmediaobjectprocessor/domain/FdoProfileAttributes.java:5]: The name is mediaObject in the Handle API. I'll change it when I do the big overhaul on handle API to align with the new RFC. 
[Jira](https://naturalis.atlassian.net/browse/DD-596)

**Review by samleeflang — APPROVED** [review]
:partying_face:

**Pull request merged**